clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/security/AmazonCredentialsLifecycleHandler.java

```diff
@@ -0,0 +1,273 @@
+import com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider;
+import com.netflix.spinnaker.clouddriver.aws.provider.agent.ImageCachingAgent;
+import com.netflix.spinnaker.clouddriver.aws.provider.agent.ReservationReportCachingAgent;
+import com.netflix.spinnaker.clouddriver.aws.provider.config.ProviderHelpers;
+import com.netflix.spinnaker.clouddriver.aws.provider.view.AmazonS3DataProvider;
+import com.netflix.spinnaker.config.AwsConfiguration.DeployDefaults;
+import com.netflix.spinnaker.credentials.CredentialsLifecycleHandler;
+import com.netflix.spinnaker.credentials.CredentialsRepository;
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+@Component
+@Lazy
+@Slf4j
+@RequiredArgsConstructor
+public class AmazonCredentialsLifecycleHandler
+    implements CredentialsLifecycleHandler<NetflixAmazonCredentials> {
+  private final AwsCleanupProvider awsCleanupProvider;
+  private final AwsInfrastructureProvider awsInfrastructureProvider;
+  private final AwsProvider awsProvider;
+  private final AmazonCloudProvider amazonCloudProvider;
+  private final AmazonClientProvider amazonClientProvider;
+  private final AmazonS3DataProvider amazonS3DataProvider;
+
+  private final AwsConfigurationProperties awsConfigurationProperties;
+  private final ObjectMapper objectMapper;
+  private final @Qualifier("amazonObjectMapper") ObjectMapper amazonObjectMapper;
+  private final EddaApiFactory eddaApiFactory;
+  private final ApplicationContext ctx;
+  private final Registry registry;
+  private final Optional<ExecutorService> reservationReportPool;
+  private final Optional<Collection<AgentProvider>> agentProviders;
+  private final EddaTimeoutConfig eddaTimeoutConfig;
+  private final DynamicConfigService dynamicConfigService;
+  private final DeployDefaults deployDefaults;
+  private final CredentialsRepository<NetflixAmazonCredentials>
+      credentialsRepository; // Circular dependency.
+  protected Set<String> publicRegions = new HashSet<>();
+  protected Set<String> awsInfraRegions = new HashSet<>();
+  protected boolean reservationReportCachingAgentScheduled = false;
+
+  @Override
+  public void credentialsAdded(@NotNull NetflixAmazonCredentials credentials) {
+    scheduleAgents(credentials);
+    scheduleReservationReportCachingAgent();
+    if (reservationReportCachingAgentScheduled) {
```

Comment:
It strikes me as weird to have widespread knowledge of specific caching agents, and having to manipulate local state in an agent even more so.

If we want to track whether an account is VPC or not, I'd think we could add an attribute to `NetflixAmazonCredentials`.

Reply:
yeah that's what I meant also in the previous comment, ie to have a flag in `NetflixAmazonCredentials` to check if the account is vpc only. maybe we can fix this in the next PR.

Reply:
Yes, I do think that is the better approach.

In this case, I'm quite sure we don't actually need to worry about vpc/non-vpc in this particular agent (I'd be super surprised if anyone but Netflix uses it).

https://github.com/spinnaker/clouddriver/pull/5045 drops the non-vpc handling and cleans up the describes we were making in the lifecycle handler.

Should something like this come up again, we can enhance `NetflixAmazonCredentials`.